tornado/concurrent.py

```diff
@@ -337,11 +337,12 @@ def _check_done(self):
 
     def _set_done(self):
         self._done = True
-        from tornado.ioloop import IOLoop
-        loop = IOLoop.current()
-        for cb in self._callbacks:
-            loop.add_callback(cb, self)
-        self._callbacks = None
+        if self._callbacks:
+            from tornado.ioloop import IOLoop
+            loop = IOLoop.current()
+            for cb in self._callbacks:
+                loop.add_callback(cb, self)
+            self._callbacks = None
 
     # On Python 3.3 or older, objects with a destructor part of a reference
```

Comment:
isn't python 3.3 deprecated?

Reply:
Yes, but python2.7 is still supported, so we can't remove this condition yet. 

(aktiger87, there's an unsubscribe button on the web view of this issue)